Apps/Examples/Examples/All Examples/Custom2DPuckExample.swift

```diff
@@ -4,27 +4,158 @@ import MapboxMaps
 @objc(Custom2DPuckExample)
 public class Custom2DPuckExample: UIViewController, ExampleProtocol {
 
-    internal let toggleAccuracyRadiusButton: UIButton = UIButton(frame: .zero)
-    internal var mapView: MapView!
-    internal var showsAccuracyRing: Bool = false {
+    private var mapView: MapView!
+    internal var puckConfiguration = Puck2DConfiguration.makeDefault(showBearing: true)
+    private var showsPuck: PuckVisibility = .isVisible {
         didSet {
-            syncPuckAndButton()
+            mapView.location.options.puckType = showsPuck.isVisible ? .puck2D(puckConfiguration) : .none
+        }
+    }
+
+    private var puckImage: PuckImage = .blueDot {
+        didSet {
+            puckConfiguration.topImage = puckImage.image
+            updatePuckUI()
+        }
+    }
+
+    private var showsBearing: PuckBearingVisibility = .isVisible {
+        didSet {
+            // It's necessary to create a new default when switching bearing visibility
+            var newPuck = Puck2DConfiguration.makeDefault(showBearing: showsBearing.isVisible)
+            newPuck.topImage = puckConfiguration.topImage
+            newPuck.showsAccuracyRing = puckConfiguration.showsAccuracyRing
+            puckConfiguration = newPuck
+            updatePuckUI()
+        }
+    }
+
+    private var showsAccuracyRing: PuckAccuracyRingVisibility = .isHidden {
+        didSet {
+            puckConfiguration.showsAccuracyRing = showsAccuracyRing.isVisible
+            updatePuckUI()
+        }
+    }
+
+    private var bearingSource: PuckBearingSource = .course {
+        didSet {
+            mapView.location.options.puckBearingSource = bearingSource
+        }
+    }
+
+    private var style: Style = .dark {
+        didSet {
+            mapView.mapboxMap.style.uri = style.styleURL
+        }
+    }
+
+    private var projection: StyleProjectionName = .mercator {
+        didSet {
+            do {
+                try mapView.mapboxMap.style.setProjection(StyleProjection(name: projection))
+            } catch {
+                print(error)
+            }
```

Comment:
Can you extract this to the separate function as well? We usually use very lightweight didSet observers.